### DISCUSSION TITLE
How do I use [StructLayout(LayoutKind.Explicit)] in f#

### DISCUSSION THREAD

**Discussion by Xyncgas** [fsharp/fslang-suggestions#1217] [General]
I know you can declare a type as a struct easily with [<Struct>] but I would also like to use explicit layout

**Comment by abelbraaksma** [comment]
There’s nothing that stops you from doing this. Just apply the necessary attributes to the fields. This will have the same rules and applications as in C# (that is, if any field references a type with auto layout, the JIT will not honour it).
